README.md

```diff
@@ -1,118 +1,144 @@
+
+#### Functional Components
+
+Functional components receive the object of component properties as their only parameter.
+
+```javascript
+function myComponent({ element, children, options }) { ... }
 ```
 
-If you are compiling your code with a bundler like webpack, only import what you need for a smaller footprint:
-```js
-import { Component } from 'js-component-framework/es';
+#### Classical Components
+
+For classical components, the object of component properties is passed as the constructor's only parameter.
+
+```javascript
+class MyComponent {
+  constructor({ element, children, options }) { ... }
+}
 ```
 
-When writing a component class, are some rules you need to follow and some guidelines:
+## Loading Components
 
-* You _must_ provide a constructor. At minimum, the constructor should look like the following:
-	```js
-	constructor(config) {
-		super(config);
-	}
-	```
-	This constructor will be called when ComponentManager instances your component, your configuration passed in, then passed to the parent Component class using the `super()` function.
-* Don't perform much logic within the class constructor. Instead, separate logic into distinct methods, each for a specific purpose. Use the constructor to call these methods and initialize any runtime component logic, event listeners, calculations, etc.
-* Any method you might call from another component or use as a callback for an event listener should be bound to the component class using `this.myMethod = this.myMethod.bind(this)`. Methods can be called from other components using the component manager via `manager.callComponentMethod`. See the ComponentManager class for documentation.
+Due to the nature of the component framework, components must be prepared for loading prior to initialization. By default, the function that performs this preparation will also load the component.
 
-[See the Header example](./examples/Header/Header.js) for context.
+### componentProvider
 
-## How to instantiate components
+`componentProvider` creates a provider function that contains all the properties and DOM querying logic necessary for all instances of the component to be initialized.
 
-Import the component config in one of your entry point files, and pass that config to the ComponentManager's instanceComponents method.
+The component will automatically be initialized according to the `config.load` value. If `config.load` is `false`, the provider function is returned.
 
-Generally speaking, you'll want to instantiate your component in the footer or on `DOMContentLoaded`.
+**Parameters**
 
-```js
+**config** _(Required)_ A component config object.
 
-import { ComponentManager } from `js-component-framework/es`;
-import headerConfig from `./Components/Header`;
+```javascript
+import { componentProvider } from 'js-component-framework';
+import productDetails from './productDetails';
+
+const productDetailsConfig = {
+  name: 'product-details',
+  component: productDetails,
```

Comment:
This suggested pattern can be a bit redundant albeit it has been the suggested pattern from this framework since the beginning. Using this example and based off of past experience with this framework, the `productDetails` imported class will only be imported here, once. Why not declare that class in this same file? Additionally this pattern may slow down build processes by increasing the number of import modules.

The developer may find themselves switching back and forth between the config and class files.

Reply:
Yeah, that's actually how I structure my components now. I was leaning toward how it's always been documents, but you make a great point – I'll update this to suggest the _Better Way_™